### PR TITLE
Adding Vagrantfiles from CDK.zip for RHEL CDK

### PR DESCRIPTION
**Pull request by LalatenduMohanty**
Signed-off-by: Lalatendu Mohanty lmohanty@redhat.com

### CODE REVIEW

**Comment by bexelbie** [comment]
Does this put everything in cdk.zip in this or other repos?  Can we maybe commit a map/listing file somewhere?

**Comment by bexelbie** [comment]
LGTM -Will let @LalatenduMohanty  merge pending response above.

File: components/rhel/misc/rhel-k8s-singlenode-setup/README.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
+Vagrantfile: Single Node Kubernetes
+===================================
+
+This Vagrantfile is the suggested configuration for using the CDK with Kubernetes.  This file sets up private networking that will be used to expose the docker daemon and kubernetes to the host.  This file also sets up a single node Kubernetes instance where the master and node are on the same machine.
```
Comment (navidshaikh) [inline components/rhel/misc/rhel-k8s-singlenode-setup/README.rst]: [..]expose the `docker` daemon and `kubernetes` [..]. This Vagrantfile also sets up a single node Kubernetes instance [..]

File: components/rhel/misc/rhel-k8s-singlenode-setup/README.rst [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+This Vagrantfile is useful for anyone using host-based tools, such as the `Eclipse docker tooling <https://wiki.eclipse.org/Linux_Tools_Project/Docker_Tooling>`_ or the kubernetes and docker CLIs, with the CDK.
```
Comment (navidshaikh) [inline components/rhel/misc/rhel-k8s-singlenode-setup/README.rst]: or the `docker`, `oc` or `kubectl` CLIs

File: components/rhel/misc/rhel-k8s-singlenode-setup/README.rst [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+QuickStart
+----------
+
+1. Get latest CDK box and add it to vagrant.
```
Comment (navidshaikh) [inline components/rhel/misc/rhel-k8s-singlenode-setup/README.rst:11]: Write down the commands for step 1.

File: components/rhel/misc/rhel-k8s-singlenode-setup/README.rst [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+2. Create a directory for the Vagrant Box
+
+  ``$ mkdir directory && cd directory``
+
+3. Use this Vagrantfile
```
Comment (navidshaikh) [inline components/rhel/misc/rhel-k8s-singlenode-setup/README.rst:17]: Again, we need a command or hyperlink for Vagrantfile.

File: components/rhel/misc/rhel-k8s-singlenode-setup/README.rst [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+4. Start the Vagrant Box
+
+  ``$ vagrant up``
+
+5. Proceed with using the CDK box.
```
Comment (navidshaikh) [inline components/rhel/misc/rhel-k8s-singlenode-setup/README.rst:23]: Either omit it or re-phrase it for some use case.

File: components/rhel/misc/rhel-k8s-singlenode-setup/README.rst [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+  You may wish to ``vagrant ssh`` into the CDK and verify that Kubernetes is setup using the ``kubectl`` CLI as follows:
+
+  ::
+
+    $ kubectl get nodes
+    NAME        LABELS                             STATUS
+    127.0.0.1   kubernetes.io/hostname=127.0.0.1   Ready
```
Comment (navidshaikh) [inline components/rhel/misc/rhel-k8s-singlenode-setup/README.rst:31]: IMO, this should be kept in step 5.

File: components/rhel/rhel-ose/README.rst [hunk 7]
```diff
@@ -0,0 +1,53 @@
+Vagrantfile to setup  OpenShift Enterprise
+==========================================
+
+This Vagrantfile sets up the CDK for development with OpenShift.  This file also sets up private networking that will be used to expose various services to the host.  This Vagrantfile also provisions an instance of `OpenShift Enterprise <https://www.openshift.com/enterprise/whats-new.html>`_.
+
+If you are interested in the process used in the Vagranfile to setup OpenShift, please read the comments at the top of the file.
```
Comment (navidshaikh) [inline components/rhel/rhel-ose/README.rst:6]: [..], please read the comments at the top of the Vagrantfile.

Also we can hyperlink the Vagrantfile

File: components/rhel/rhel-ose/README.rst [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+QuickStart
+----------
+
+1. Get latest CDK box (Using Developer Subscription) and add it to vagrant per `Installing the Atomic Developer Bundle <../../../docs/installing.rst>`_.
```
Comment (navidshaikh) [inline components/rhel/rhel-ose/README.rst]: this looks good, or probably add the commands directly here.
You can use the same pattern in kube vagrantfile as I commented https://github.com/projectatomic/adb-atomic-developer-bundle/pull/263/files#r54400774

File: components/rhel/rhel-ose/README.rst [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+2. Create a directory for the Vagrant Box
+
+  ``$ mkdir directory && cd directory``
+
+3. Download this Vagrantfile in same directory.
+
```
Comment (navidshaikh) [inline components/rhel/rhel-ose/README.rst:18]: two suggestion,
1 - Hyperlink to Vagrantfile
2 - Write down the command

File: components/rhel/rhel-ose/README.rst [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+  **Note:** The CDK OpenShift Vagrant Box IP defaults to *10.1.2.2*. If you want a different IP then change the **PUBLIC_ADDRESS** variable in the Vagrantfile.
+
+4. Start the Vagrant Box
+
+  ``$ vagrant up``
+
+5. Proceed with using the CDK.
+
```
Comment (navidshaikh) [inline components/rhel/rhel-ose/README.rst:26]: Either remove step 5 or write down use case, for example checking OpenShift status after vagrant ssh.

**Comment by LalatenduMohanty** [comment]
@bexelbie 

> Does this put everything in cdk.zip in this or other repos? Can we maybe commit a map/listing file somewhere?

CDK.zip contains these Vagrantfiles and ruby gems. +1 for having a list/map somewhere in the documentation

**Comment by bexelbie** [comment]
@LalatenduMohanty can you add that mapping, perhaps as a README in the rhel directory?

File: components/rhel/misc/rhel-k8s-singlenode-setup/README.rst (hunk 3 above)
Comment (LalatenduMohanty) [inline components/rhel/misc/rhel-k8s-singlenode-setup/README.rst:11]: @navidshaikh nope, this is not required. Lets not try to write documentation for CDK. In future we can just point to the documentation when it is available.

File: components/rhel/misc/rhel-k8s-singlenode-setup/README.rst (hunk 6 above)
Comment (LalatenduMohanty) [inline components/rhel/misc/rhel-k8s-singlenode-setup/README.rst:31]: @navidshaikh `kubectl get nodes` is an optional thing. So it does not have to be listed as a compulsory step.

File: components/rhel/rhel-ose/README.rst (hunk 9 above)
Comment (LalatenduMohanty) [inline components/rhel/rhel-ose/README.rst:18]: @navidshaikh We are not trying to create documentation for CDK.  As I mentioned before in future we can just point to the relevant documentation when it is published.

**Comment by LalatenduMohanty** [comment]
@bexelbie I have added a mapping in the README. Let me know if this is ok.

File: components/rhel/rhel-ose/README.rst (hunk 8 above)
Comment (LalatenduMohanty) [inline components/rhel/rhel-ose/README.rst]: @navidshaikh I have removed the link to ADB documentation in the PR as we will have CDK documentation in Red Hat access.redhat.com and users should follow that.

**Comment by bexelbie** [comment]
@LalatenduMohanty we need to expand hte map to list every file that lands in that zip and where it is sourced from, including READMEs.  Those READMEs need to be listed so they can be found and edited.  This is a situation where a downstream build has docs living somewhere other than on their doc site.

**Comment by LalatenduMohanty** [comment]
@bexelbie Agree. Will update the PR

**Comment by bexelbie** [comment]
@LalatenduMohanty What is the status of this?  RH Docs would like to be able to change their READMEs and can't do so easily until they land here.

**Comment by LalatenduMohanty** [comment]
@bexelbie The READMEs are already present in respective upstream projects. I don't have plan to bring them here but create a map where others can relate and find them easily.

**Comment by praveenkumar** [comment]
@LalatenduMohanty I think @bexelbie is asking about top level README which is still not part of this patch. Upstream README we should directly map.

```
Red Hat Container Development Kit v2.0 Beta 5
===================================

Red Hat Container Development Kit (CDK) is a set of tools that helps you quickly and easily set up an environment for developing and testing containerized applications on the Red Hat Enterprise Linux platform. Red Hat CDK is available for Microsoft Windows, Mac OS X, and Linux operating systems, thus allowing you to use your favorite platform to produce applications that are ready to deploy in the Red Hat Enterprise Linux ecosystem.

Red Hat CDK is a part of the Red Hat Developers program, which provides tools, resources, and support for developers who wish to utilize Red Hat solutions and products to create applications, both locally and in the cloud. For additional information and to register to become a part of the program, visit [developers.redhat.com](http://developers.redhat.com). 

Installation Instructions
=========================
Detailed installation instructions for setting up Red Hat CDK on each of the supported platforms (Microsoft Windows, Mac OS X, and Linux) are available in the [CDK Installation Guide](https://access.redhat.com/documentation/en/red-hat-enterprise-linux-atomic-host/version-7/container-development-kit-guide)

Getting Started
===============
The [Getting Started Guide](https://access.redhat.com/documentation/en/red-hat-enterprise-linux-atomic-host/version-7/getting-started-with-container-development-kit/) describes the individual components that form Red Hat CDK, followed by instructions on how to obtain, install, and configure the components, so that you can start using the resulting environment to interact with containers and develop containerized applications.

An additional document, the Container Development Guide, will be published with The GA release of the CDK. This guide contains practical usage examples, including advanced topics such as developing multi-container applications and container orchestration using Kubernetes or OpenShift.

Updated Releases
================
Updated releases of the CDK will be available to download from the Red Hat Customer portal, [access.redhat.com](https://access.redhat.com/) under Red Hat Enterprise Linux. Select [Red Hat Container Development Kit](https://access.redhat.com/downloads/content/293/ver=2/rhel---7/2.0.0/x86_64/product-software) from the list of product variants.


Beta Feedback and Questions
===========================
Please join the [container-tools@redhat.com](container-tools@redhat.com) mailing list to ask questions and provide feedback.
```

**Comment by bexelbie** [comment]
@praveenkumar yes that one too

@LalatenduMohanty some of the upstream README's reflect things that are not supported by the RH CDK.  RH needs to be able to create a custom version of those that omits the unsupported material.

**Comment by LalatenduMohanty** [comment]
@bexelbie @praveenkumar This patch is for the Vagrantfiles. Lets keep it that way. We should have separate patches for other stuff you are suggesting. I can create the mapping in this patch but we cant just keep on adding new requirements to this patch.

**Comment by bexelbie** [comment]
@LalatenduMohanty that is fine, we will get the new files in, but this patch needs to be complete and land as it is now blocking on multiple levels.

**Comment by LalatenduMohanty** [comment]
@bexelbie This PR is complete except the mapping right? Or am I missing something else?

**Comment by LalatenduMohanty** [comment]
@bexelbie Please review. I think I have added the required bits.

**Comment by LalatenduMohanty** [comment]
@bexelbie @praveenkumar @navidshaikh ping. IMO this is ready to be merged.

**Comment by navidshaikh** [comment]
LGTM. @bexelbie should give an ACK and merge.

**Comment by bexelbie** [comment]
LGTM

**Pull request merged**